packages/dev/core/src/assetContainer.ts

```diff
@@ -1281,4 +1282,79 @@ export class AssetContainer extends AbstractAssetContainer {
 
         this.populateRootNodes();
     }
+
+    /**
+     * Get from a list of objects by tags
+     * @param list the list of objects to use
+     * @param tagsQuery the query to use
+     * @param filter a predicate to filter for tags
+     * @returns
+     */
+    private _getByTags<T>(list: T[], tagsQuery: string, filter?: (item: T) => boolean): T[] {
+        if (tagsQuery === undefined) {
+            // returns the complete list (could be done with Tags.MatchesQuery but no need to have a for-loop here)
+            return list;
+        }
+
+        const listByTags = [];
+
+        for (const i in list) {
+            const item = list[i];
+            if (Tags && Tags.MatchesQuery(item, tagsQuery) && (!filter || filter(item))) {
+                listByTags.push(item);
+            }
+        }
+
+        return listByTags;
+    }
+
+    /**
+     * Get a list of meshes by tags
+     * @param tagsQuery defines the tags query to use
+     * @param filter defines a predicate used to filter results
+     * @returns an array of Mesh
+     */
+    public getMeshesByTags(tagsQuery: string, filter?: (mesh: AbstractMesh) => boolean): AbstractMesh[] {
+        return this._getByTags(this.meshes, tagsQuery, filter);
+    }
+
+    /**
+     * Get a list of cameras by tags
+     * @param tagsQuery defines the tags query to use
+     * @param filter defines a predicate used to filter results
+     * @returns an array of Camera
+     */
+    public getCamerasByTags(tagsQuery: string, filter?: (camera: Camera) => boolean): Camera[] {
+        return this._getByTags(this.cameras, tagsQuery, filter);
+    }
+
+    /**
+     * Get a list of lights by tags
+     * @param tagsQuery defines the tags query to use
+     * @param filter defines a predicate used to filter results
+     * @returns an array of Light
+     */
+    public getLightsByTags(tagsQuery: string, filter?: (light: Light) => boolean): Light[] {
+        return this._getByTags(this.lights, tagsQuery, filter);
+    }
+
+    /**
+     * Get a list of materials by tags
+     * @param tagsQuery defines the tags query to use
+     * @param filter defines a predicate used to filter results
+     * @returns an array of Material
+     */
+    public getMaterialsByTags(tagsQuery: string, filter?: (material: Material) => boolean): Material[] {

```

Comment:
@sebavan the function is called `getMaterialByTags` (missing an "s") in `scene.ts`.
Sounds like a typo to me, but changing it in the scene would be a breaking change.

Reply:
yup lets not change it in scene :-)